src/node.ts

```diff
@@ -0,0 +1,65 @@
+// This file attaches all functionality to Node
+// that would otherwise require circular dependencies.
+import Node from './node-internal'
+import Collection from './collection'
+import Literal from './literal'
+import { ValueType } from './types'
+
+export default Node
+
+/**
+ * Creates an RDF Node from a native javascript value.
+ * RDF Nodes are returned unchanged, undefined returned as itself.
+ * @method fromValue
+ * @param value - Any native Javascript value
+ */
+Node.fromValue = function (value: ValueType): Node | Literal | undefined | null | Collection {
+  if (typeof value === 'undefined' || value === null) {
+    return value
+  }
+  const isNode = Object.prototype.hasOwnProperty.call(value, 'termType')
+  if (isNode) {  // a Node subclass or a Collection
+    // @ts-ignore
```

Comment:
I'd preferably avoid adding `@ts-ignore` if possible, but if not, please at least add a comment indicating why it is required here, e.g.

    // @ts-ignore: Cannot cast `value` to the the required type because...

I did not add a comment to every `@ts-ignore`, as I'm sure you can do a Find in all Files ;)